src/hooks/useMissionsMaxAPY.ts

```diff
@@ -1,67 +1,119 @@
-const MERKL_API = 'https://api.merkl.xyz/v3';
-const CREATOR_TAG = MERKL_CREATOR_TAG;
+
+interface MerklV4Token {
+  address: string;
+  symbol: string;
+  decimals: number;
+  price: number;
+}
+
+interface MerklV4Opportunity {
+  chainId: number;
+  type: string;
+  identifier: string;
+  name: string;
+  status: string;
+  apr: number;
+  dailyRewards: number;
+  tokens: MerklV4Token[];
+  aprRecord: {
+    cumulated: number;
+    timestamp: string;
+    breakdowns: Array<{
+      distributionType: string;
+      identifier: string;
+      type: string;
+      value: number;
+    }>;
+  };
+  rewardsRecord: {
+    total: number;
+    timestamp: string;
+    breakdowns: Array<{
+      token: MerklV4Token;
+      amount: string;
+      value: number;
+      distributionType: string;
+      campaignId: string;
+    }>;
+  };
+}
 
 interface useMissionsAPYRes {
   isLoading: boolean;
   isSuccess: boolean;
   apy: number;
+  data: MerklV4Opportunity[];
 }
 
 export const useMissionsMaxAPY = (
-  claimingIds: string[] | undefined,
+  searchQuery: string[] | undefined,
+  chainIds: number[] | undefined,
 ): useMissionsAPYRes => {
-  const MERKL_CAMPAIGN_API = `${MERKL_API}/campaigns?chainIds=${ACTIVE_CHAINS.join(',')}`; //&creatorTag=${CREATOR_TAG}`;
+  // Create all combinations of chainIds and searchQuery with simple underscore check
+  const queryCombinations = (chainIds || ACTIVE_CHAINS).flatMap((chainId) =>
+    (searchQuery || []).map((claimingId) => ({
+      chainId,
+      address: claimingId.includes('_') ? claimingId.split('_')[1] : claimingId,
+    })),
+  );
 
-  const { data, isSuccess, isLoading } = useQuery({
-    queryKey: ['accountCampaignInfo'],
-    queryFn: async () => {
-      try {
-        const response = await fetch(MERKL_CAMPAIGN_API);
-        const result = await response.json();
-        return result;
-      } catch (err) {
-        console.error(err);
-      }
-    },
-    enabled: claimingIds && claimingIds.length > 0,
-    refetchInterval: 1000 * 60 * 60,
+  const results = useQueries({
+    queries: queryCombinations.map(({ chainId, address }) => ({
+      queryKey: ['merklOpportunities', chainId, address],
+      queryFn: async () => {
+        try {
+          const response = await fetch(
+            `${MERKL_API}/opportunities?chainId=${chainId}&search=${address}`,
+          );
+          const result = await response.json();
+          return result as MerklV4Opportunity[];
+        } catch (err) {
+          console.error(
+            `Error fetching Merkl opportunities for chain ${chainId} and address ${address}:`,
+            err,
+          );
+          return [];
+        }
+      },
+      enabled: !!searchQuery && searchQuery.length > 0,
+      refetchInterval: 1000 * 60 * 60, // 1 hour
+      retry: 3,
+      refetchOnWindowFocus: false,
+      refetchOnMount: true,
+    })),
   });
 
+  const data = results
+    .map((r) => r.data)
+    .filter((d): d is MerklV4Opportunity[] => Array.isArray(d))
```

Comment:
you should not have to specify the type of d as it should be typed already, + avoid single letter variable name to increase readability